Allow additional rules for use opencryptoki

Koncpa opened the pull request:
Commit 1:
The var.lib.opencryptoki.* files are located in /dev/shm/ are now labeled as
pkcs_slotd_tmpfs_t.
New interface pkcs_tmpfs_filetrans() allow create and manage objects in
the tmpfs directories with a private type (pkcs_slotd_tmpfs_t) and also
allow manage tmpfs dirs.
In pkcs_use_opencryptoki add permission allowing domain to tcp connect
tcs port.

Commit 2:
Certmonger and Ipsec need more permission for proper usage of opencryptoki.
Allow certmonger_t to get attributes of tmpfs (fs_getattr_tmpfs) and also certmonger allow
create and manage objects in the tmpfs directories with a private type
pkcs_slotd_tmpfs_t (pkcs_tmpfs_filetrans). Also certmonger can proper
use opencryptoki (pkcs_use_opencryptoki).

Allow ipsec_t change owner of file via chown capability.
Also ipsec allow create and manage objects in the tmpfs directories
with a private type pkcs_slotd_tmpfs_t (pkcs_tmpfs_filetrans).


COPR build: https://download.copr.fedorainfracloud.org/results/pkoncity/selinux-policy/fedora-rawhide-x86_64/02052708-selinux-policy/
Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1894132

WOnder93 commented:
I think it would be better to use a named transition specifically for `var.lib.opencryptoki.tpm.root`, otherwise this could result in files e.g. in `/tmp` to get an unexpected label. Hopefully the services won't run as non-root in the near future, at least until I implement prefix/suffix-based filename transitions :)

Also, it would be nicer to split this into multiple commits (I would separate out at least the capability addition and the corenet_tcp_connect_tcs_port() call).

And please don;t forget to double-check the whitespace consistency in your patches ;P

zpytela commented:
The content looks good, I'd suggest escape dots in the file context specification, and use tabs for indentation in the new interface.
Consider splitting the commit and add reasons for the new permissions when they are available.
Please also change the commit message to make it more clear and comprehensible.

zpytela commented:
> I think it would be better to use a named transition specifically for `var.lib.opencryptoki.tpm.root`, otherwise this could result in files e.g. in `/tmp` to get an unexpected label. Hopefully the services won't run as non-root in the near future, at least until I implement prefix/suffix-based filename transitions :)

This is a transition for `/dev/shm` with the `tmpfs_t` type, so why are you worried about `/tmp`?

Koncpa committed:
Label var.lib.opencryptoki.* files and create pkcs_tmpfs_filetrans()

The var.lib.opencryptoki.* files are located in /dev/shm/
are now labeled as pkcs_slotd_tmpfs_t.
New interface pkcs_tmpfs_filetrans() allow
create and manage objects in the tmpfs directories with a private
type (pkcs_slotd_tmpfs_t) and also allow manage tmpfs dirs.
In pkcs_use_opencryptoki add permission allowing
domain to tcp connect tcs port.

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1894132

zpytela commented on policy/modules/contrib/pkcs.if:
```diff
@@ -116,6 +116,29 @@ interface(`pkcs_getattr_exec_files',`
+#
+interface(`pkcs_tmpfs_filetrans',`
+	gen_require(`
+        type pkcs_slotd_tmpfs_t;
```
Please replace 8 spaces with 2 tabs.

zpytela commented:
Last two requests:
- split the "ipsec and certmonger" commit into two to ease backporting
- add a bugzilla reference to each commit

Koncpa committed:
Allow using opencryptoki for certmonger

Allow certmonger_t to get attributes of tmpfs (fs_getattr_tmpfs)
and also certmonger allow create and manage objects in the tmpfs
directories with a private type pkcs_slotd_tmpfs_t (pkcs_tmpfs_filetrans).
Also certmonger can proper use opencryptoki (pkcs_use_opencryptoki).

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1894132

Koncpa committed:
Allow using opencryptoki for ipsec

Allow ipsec_t change owner of file via chown capability.
Also ipsec allow create and manage objects in the tmpfs directories
with a private type pkcs_slotd_tmpfs_t (pkcs_tmpfs_filetrans).

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1894132

zpytela commented:
LGTM now, thank you.
